test/metabase/mbql/generate.clj

```diff
@@ -0,0 +1,88 @@
+(ns metabase.mbql.generate
+  (:require
+   [clojure.test :as t]
+   [clojure.test.check.generators :as gens]
+   [metabase.mbql.generate.aggregations :as gen.aggregations]
+   [metabase.mbql.generate.common :as gen.common]
+   [metabase.mbql.generate.data :as gen.data]
+   [metabase.mbql.generate.expressions :as gen.expressions]
+   [metabase.query-processor :as qp]
+   [metabase.test :as mt]
+   [metabase.util :as u]
+   [toucan.db :as db]))
+
+(defn fields-generator [field-generator]
+  (gens/such-that seq (gens/vector-distinct field-generator)))
+
+(def limit-generator
+  (gens/large-integer* {:min 1}))
+
+(defn order-by-generator [field-generator]
+  ;; TODO -- order by expression references or aggregation references. I don't think arbitary expressions are currently
+  ;; allowed.
+  (gens/such-that seq
+                  (gens/vector-distinct (gens/let [field field-generator
+                                                   direction (gens/elements #{:asc :desc})]
+                                          [direction field]))
+                  1000))
+
+(defn inner-query-generator [field-generator]
+  (gens/let [expressions (gens/one-of [(gens/return nil)
+                                       (gen.expressions/expressions-map-generator field-generator)])]
```

Comment:
Consider using `gens/frequency` instead of `gens/one-of` unless `nil` is the typical value.